src/main/java/com/icthh/xm/ms/entity/domain/Event.java

```diff
@@ -86,116 +103,57 @@
     @JsonIdentityReference(alwaysAsId = true)
     private XmEntity assigned;
 
-    public Long getId() {
-        return id;
-    }
-
-    public void setId(Long id) {
-        this.id = id;
-    }
-
-    public String getTypeKey() {
-        return typeKey;
-    }
+    /**
+     * Reference to {@link XmEntity} which stores extra data regarding to this {@link Event}.
+     */
+    @ApiModelProperty(value = "Reference to event's extra data")
+    @OneToOne(cascade = {PERSIST, MERGE, REMOVE})
+    @JoinColumn(name = "event_data_ref_id", unique = true)
+    private XmEntity eventDataRef;
```

Comment:
when we delete this XmEntity what will be happend with event

Reply:
typeKey of this entity must be specified in specification for this event, and validate here

Reply:
will be a foreign key constraint violation

Reply:
> when we call delete this XmEntity what will be happend with this event

should Event be deleted in this case or just delete XmEntity and unlink it with related Event?

Reply:
> 1. add entity type to event definition in specification and add validation for this typeKey
> 2. handle delete entity when it has ref in event
1. need additional internal conversation
2. done